### PR TITLE
Change Pytorch materializer to support remote storage

### PR DESCRIPTION
**Pull request by safoinme**
## Describe changes
change PyTorch materializer input & output handler to support remote storage using zenml.io fileio context manager.
this will allow saving model artifacts directly to any storage system.

```
        with fileio.open(
            os.path.join(self.artifact.uri, DEFAULT_FILENAME), "wb"
        ) as f:
            torch.save(model, f)
```
I have added another save for the model checkpoint which is essential for the inference phase.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by htahir1 — APPROVED** [review]
If you have tested it works on remote backends then I'm happy to approve. Would be cool if you could test on GCP and AWS at least and Azure would be a bonus :-)

**Comment by safoinme** [comment]
> If you have tested it works on remote backends then I'm happy to approve. Would be cool if you could test on GCP and AWS at least and Azure would be a bonus :-)

@htahir1  I have added an other save in pytorch materializer to save entire_model and checkpoint. using the entire model cause error in inference (something related to torch.load) 
for testing, I have done it for GCP and it's working. I'm almost done with Pytorch custom deployment with Seldon once it's finished I will ask @stefannica to test it in AWS since it's his main stack I think.

File: src/zenml/integrations/pytorch/materializers/pytorch_materializer.py [hunk 1]
```diff
@@ -49,4 +52,21 @@ def handle_return(self, model: Union[Module, TorchDict]) -> None:
...
+
+        # Save model checkpoint to artifact directory, This is the default behavior for loading model in production phase (inference)
+        if isinstance(model, Module):
+
+            with fileio.open(
+                os.path.join(self.artifact.uri, CHECKPOINT_FILENAME), "wb"
+            ) as f:
+                torch.save(model.state_dict(), f)
+
+        else:
```
Comment (bcdurak) [inline src/zenml/integrations/pytorch/materializers/pytorch_materializer.py:70]: If we don't need to do anything here, could we perhaps remove the last two lines?

**Review by bcdurak — APPROVED** [review]
Great changes. I left a tiny comment, but other than that everything looks good to me!

File: src/zenml/integrations/pytorch/materializers/pytorch_materializer.py (hunk 1 above)
Comment (safoinme) [reply src/zenml/integrations/pytorch/materializers/pytorch_materializer.py]: yes sure, it's removed!

**Pull request merged**